docs/tutorials/hyperparameter_tuning/hp_tuning_with_optuna.md

```diff
@@ -0,0 +1,328 @@
+        self.freq = freq
+        self.metric_type = metric_type
+    
+    def get_params(self, trial) -> dict:
+        return {
+            "num_layers": trial.suggest_int("num_layers", 1, 5),
+            "hidden_size": trial.suggest_int("hidden_size", 10, 50),
+        }
+
+    def split_entry(self, entry):
+        entry_past = {}
+        for key, value in entry.items():
+            if key == "target":
+                entry_past[key] = value[: len(value) - self.prediction_length]
+            else:
+                entry_past[key] = value
+
+        df = pd.DataFrame(entry['target'], columns=[entry['item_id']])
+        df = df.set_index(pd.date_range(start=entry['start'], 
+                                        periods=len(entry['target']), freq=self.freq))
+        return entry_past, df[-self.prediction_length:]
+     
+    def __call__(self, trial):
+        params = self.get_params(trial)
+        estimator = DeepAREstimator(
+            num_layers=params['num_layers'],
+            hidden_size=params['hidden_size'],
+            prediction_length=self.prediction_length, 
+            freq=self.freq,
+            trainer_kwargs={
+                "progress_bar_refresh_rate": 0, 
+                "weights_summary": None, 
+                "max_epochs": 5, 
+            }
+        )
+        
+        entry_splited = [self.split_entry(entry) for entry in self.dataset]
+        entry_pasts = [entry[0] for entry in entry_splited]
+        entry_futures = [entry[1] for entry in entry_splited]
+        
+        predictor = estimator.train(entry_pasts, cache_data=True)
+        forecast_it = predictor.predict(entry_pasts)
+        
+        forecasts = list(forecast_it)
+        tss = list(entry_futures)
+        
+        evaluator = Evaluator(quantiles=[0.1, 0.5, 0.9])
+        agg_metrics, item_metrics = evaluator(iter(tss), iter(forecasts), num_series=len(self.dataset))
+        return agg_metrics[self.metric_type]
+```
+
+Implement the Optuna tuning process.
+
+
+```python
+import time
+start_time = time.time()
+study = optuna.create_study(direction="minimize")
+study.optimize(Objective(electricity_train_sub, dataset.metadata.prediction_length, dataset.metadata.freq),
+               n_trials=10)
+
+print("Number of finished trials: {}".format(len(study.trials)))
+
+print("Best trial:")
+trial = study.best_trial
+
+print("  Value: {}".format(trial.value))
+
+print("  Params: ")
+for key, value in trial.params.items():
+    print("    {}: {}".format(key, value))
+print(time.time() - start_time)
+```
+
+# Training deepar estimator
+
+After getting the best hyperparameters by optuna,you can set them into the DeepAR estimator to realize the prediction of electricity dataset.
+
+## Training and predict
+
+Now We can begin with GulonTS's pre-built DeepAR estimator after tuning its hyperparameters by optuna. The next process consists of training a model, producing forecasts, and evaluating the results.
```

Comment:
I think this sentence is not clear: what do you mean here? Do you mean that now we can retrain the model on the training data using the optimal hyperparameters?

Reply:
Yes，Lorenzo!I mean that.After we tune the gulonTS model and get the optimal hyperparameters we can retrain the model and evaluate  our forecast.